.github/workflows/docs.yaml

```diff
@@ -4,50 +4,71 @@ on:
   push:
     branches:
       - master
-      - v*
   pull_request:
 
 jobs:
-  test-docs:
-    name: Test Docs
+  test-or-deploy:
+    name: ${{ github.event_name == 'push' && 'Deploy' || 'Test' }}
     runs-on: ubuntu-latest
+    container: debian:bookworm
     steps:
-      - uses: actions/checkout@v3
-      - uses: actions/setup-node@v3
+      - name: Install Dependencies
+        run: |
+          apt-get update
+          apt-get -y install \
+            build-essential \
+            ca-certificates \
+            flatbuffers-compiler-dev \
+            g++-10 \
+            gcc-10 \
+            git \
+            gnupg2 \
+            jq \
+            libflatbuffers-dev \
+            libfmt-dev \
+            libpcap-dev tcpdump \
+            libsimdjson-dev \
+            libspdlog-dev \
+            libssl-dev \
+            libunwind-dev \
+            libyaml-cpp-dev \
+            libxxhash-dev \
+            lsb-release \
+            ninja-build \
+            pkg-config \
+            python3-dev \
+            python3-pip \
+            python3-venv \
+            wget \
+            yarnpkg
+          python3 -m pip install --upgrade pip
+          python3 -m pip install --upgrade cmake
```

Comment:
This feels pretty heavy-weight...why?

Reply:
This is just copy-pasted from another workflow, we actually only need some subset of this but I wanted to get this going first. I'll simplify this before asking for a review.